plugins/bookmarks/ChromiumBookmarksSearchEngine.ts

```diff
@@ -0,0 +1,109 @@
+import { existsSync, readdirSync, statSync } from 'node:fs'
+import { readFile } from 'node:fs/promises'
+import { join as joinPath } from 'path'
+
+import BookmarksSearchEngine from './BookmarksSearchEngine'
+import { Bookmark } from './types'
+
+// fields we are interested in
+// See BookmarkTreeNode in the below reference
+// https://developer.chrome.com/docs/extensions/reference/bookmarks/#type-BookmarkTreeNode
+type ChromiumBookmarkTreeNode = {
+	type: 'url'
+	name: string
+	url: string
+} | {
+	type: 'folder'
+	name: string
+	children: ChromiumBookmarkTreeNode[]
+}
+
+// glob: dataDir/*/Bookmarks
+const findBookmarksFiles = async (dataDir: string) => {
+	if (!existsSync(dataDir) || !statSync(dataDir).isDirectory()) {
+		return []
+	}
+
+	const probeProfiles = readdirSync(dataDir, { withFileTypes: true })
+		.filter(dirent => dirent.isDirectory())
+		.map(({ name }) => joinPath(dataDir, name))
+
+	return probeProfiles
```

Comment:
suggestion: we previously had these logs to indicate which bookmarkfiles are processed, I'd like to keep them.

```
		console.log(`Found bookmark file(s): ${bookmarkFiles}`)
		console.log(`${this.bookmarks.length} bookmark(s) loaded`)
```